File: Algorithm/Programmers/Level2/solution(3).py
# 피보나치 수
### 2 이상의 n이 입력되었을 때, n번째 피보나치 수를 1234567으로 나눈 나머지를 리턴하는 함수, solution을 완성해 주세요.

def solution(n):
    answer = 0
    F = [0,1]
    
    if n < 2 :
        answer = F[n]
    else:
        for i in range(2, n+1):
            F.append(F[i-1] + F[i-2])
        answer = F[n]%1234567

    return answer

File: Algorithm/Programmers/Level2/solution(3).java
// 피보나치 수
// n번째 수를 1234567로 나눈 나머지

class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] fibonacci = new int[n+1];
        
        if (n == 0)
            return 0;
        else if (n == 1)
            return 1;
        else {
            fibonacci[0] = 0;
            fibonacci[1] = 1;
            for (int i = 2; i < n+1; i++){
                fibonacci[i] = (fibonacci[i-1] + fibonacci[i-2])%1234567;
            }
            answer = fibonacci[n];
        }
        
        
        
        return answer;
    }
}